src/modm/processing/fiber/fiber.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include "context.hpp"
+#include <modm/debug/logger.hpp>
+
+#include <functional>
+
+namespace modm {
+
+class Fiber;
+modm_always_inline void yield();
+
+namespace fiber {
+
+class Scheduler;
+
+template<int size>
+class Stack {
+  static_assert(size % 4 == 0, "Stack size must be multiple of 4.");
+  static_assert(size >= 20, "Stack size must at least 20 bytes.");
```

Comment:
Random number chosen by fair dice? The lower bound would be the size for storing the swapped register state no?

Reply:
Did some research, changed random number to 40 and added an explanation in the example file.